core/Application/Kernel/GlobalSettingsProvider/IniSettingsProvider.php

```
@@ -0,0 +1,113 @@
+     * @param string|null $pathLocal Path to the config.ini.php file. Or null to use the default.
+     * @param string|null $pathCommon Path to the common.ini.php file. Or null to use the default.
+     */
+    public function __construct($pathGlobal = null, $pathLocal = null, $pathCommon = null)
+    {
+        $this->pathGlobal = $pathGlobal ?: Config::getGlobalConfigPath();
+        $this->pathCommon = $pathCommon ?: Config::getCommonConfigPath();
+        $this->pathLocal = $pathLocal ?: Config::getLocalConfigPath();
+
+        $this->iniFileChain = new IniFileChain();
+        $this->reload();
+    }
+
+    public function reload($pathGlobal = null, $pathLocal = null, $pathCommon = null)
+    {
+        $this->pathGlobal = $pathGlobal ?: $this->pathGlobal;
+        $this->pathCommon = $pathCommon ?: $this->pathCommon;
+        $this->pathLocal = $pathLocal ?: $this->pathLocal;
+
+        $this->iniFileChain->reload(array($this->pathGlobal, $this->pathCommon), $this->pathLocal);
+    }
+
+    public function &getSection($name)
+    {
+        $section =& $this->iniFileChain->get($name);
+        return $section;
+    }
+
+    public function setSection($name, $value)
+    {
+        $this->iniFileChain->set($name, $value);
+    }
+
+    public function getIniFileChain()
+    {
+        return $this->iniFileChain;
+    }
+
+    public function getPathGlobal()
+    {
+        return $this->pathGlobal;
+    }
+
+    public function getPathLocal()
+    {
+        return $this->pathLocal;
+    }
+
+    public function getPathCommon()
+    {
+        return $this->pathCommon;
+    }
+
+    public static function getSingletonInstance($pathGlobal = null, $pathLocal = null, $pathCommon = null)
+    {
+        if (self::$instance === null) {
+            self::$instance = new IniSettingsProvider($pathGlobal, $pathLocal, $pathCommon);
+        }
+
+        return self::$instance;
```

Is there a risk that this is called with non-null parameters when the instance already exists? In that case should we throw an exception? Because the caller expect the config to be created with the provided files but in reality it is already cleated with (potentially) other files.